src/biogeophys/SurfaceWaterMod.F90

```diff
@@ -472,15 +472,25 @@ subroutine QflxH2osfcSurf(bounds, num_hydrologyc, filter_hydrologyc, &
        c = filter_hydrologyc(fc)
 
        if (h2osfcflag==1) then
-          if (frac_h2osfc_nosnow(c) <= params_inst%pc) then
-             frac_infclust=0.0_r8
-          else
-             frac_infclust=(frac_h2osfc_nosnow(c)-params_inst%pc)**params_inst%mu
-          endif
+	   ! for rice we set the runoff at 0
+         if (col%itype(c)==(200+cft_lb+46) .or. col%itype(c) == (200+cft_lb+47)) then
```

Comment:
This will need to use the standard CLM irrigated rice types, `nrice` and `nirrig_rice`. (Although why did you include rainfed rice?)

Reply:
Hey Sam, this one is a bit complicated. I guess I would need to use col%patch(bounds%begp) as there is only one patch for each irrigated crop column right? Will try and test it.

Reply:
I think `bounds%begp` refers to the first patch on the processor, in which case that wouldn't work. To get the crop type associated with a column, here's an adaptation on something I've done before (although I'd hope there's a more elegant way)

```fortran
vegtype_thiscol = 0
if (col%itype(c) == istcrop) then
    n_noncrop = 0
    do p = col%patchi(c),col%patchf(c)
        if (patch%active(p)) then
            if (patch%itype(p) >= npcropmin) then
                if (vegtype_thiscol > 0) then
                    call endrun('ERROR multiple active crop patches found in this column')
                end if
                vegtype_thiscol = patch%itype(p)
            else
                n_noncrop = n_noncrop + 1
            end if
        end if
    end do
    
    ! Check
    if (n_noncrop > 0 .and. vegtype_thiscol > 0) then
        call endrun('ERROR Active crop and non-crop patches found in this active column')
    end if
end if
```